tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala

```diff
@@ -1554,6 +1556,373 @@ trait EmailRecoveryActionSetMethodContract {
+           |			},
+           |			"c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(responseOfAndreRequest)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |	"EmailRecoveryAction/set",
+           |	{
+           |		"notUpdated": {
+           |			"$taskIdOfBob": {
+           |				"type": "notFound",
+           |				"description": "Task not found"
+           |			}
+           |		}
+           |	},
+           |	"c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def updateShouldReturnNotUpdatedWhenUnknownProperty(): Unit = {
+    val taskId: String = newCreationSetRequestAndGetTaskId()
+
+    val response: String = `given`
+      .body(
+        s"""{
+           |	"using": [
+           |		"urn:ietf:params:jmap:core",
+           |		"com:linagora:params:jmap:messages:vault"
+           |	],
+           |	"methodCalls": [
+           |		[
+           |			"EmailRecoveryAction/set",
+           |			{
+           |				"update": {
+           |					"$taskId": {
+           |						"status": "canceled",
+           |						"redundant": "will get rejected"
+           |					}
+           |				}
+           |			},
+           |			"c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |	"EmailRecoveryAction/set",
+           |	{
+           |		"notUpdated": {
+           |			"$taskId": {
+           |				"type": "invalidArguments",
+           |				"description": "Unknown properties: redundant"
+           |			}
+           |		}
+           |	},
+           |	"c1"
+           |]""".stripMargin)
+  }
+
+  def newCreationSetRequestAndGetTaskId(): String = {
+    `given`
+      .body(
+        s"""{
+           |	"using": [
+           |		"urn:ietf:params:jmap:core",
+           |		"com:linagora:params:jmap:messages:vault"
+           |	],
+           |	"methodCalls": [
+           |		[
+           |			"EmailRecoveryAction/set",
+           |			{
+           |				"create": {
+           |					"clientId1": {
+           |						"subject": "subject contains"
+           |					}
+           |				}
+           |			},
+           |			"c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .extract()
+      .jsonPath()
+      .get("methodResponses[0][1].created.clientId1.id").toString
+  }
+
   def templateDeletedMessage(messageId: MessageId = randomMessageId,
                              mailboxId: MailboxId,
                              user: Username = BOB,
```

Comment:
Should we add a test with the get method to check that a task got updated correctly after a set update for example? WDYT?

Reply:
I'm thinking how to mock task Manager in this class, The fact is the test run fast, and difficult to make it waiting -> canceled
Mockito is not a good idea, because we have another test

Any idea?

Reply:
Good point hard to test... Maybe a test with the get contract just doing a assertUntil the task is comleted should be fine. I'm ok if it's too complex to skip it.